common/djangoapps/third_party_auth/models.py

```diff
@@ -882,10 +882,19 @@ class Meta(object):
 
 class ProviderApiPermissions(models.Model):
     """
-    This model links OAuth2 client with provider Id.
+    DEPRECATED: This model links OAuth2 client with provider Id.
 
     It gives permission for a OAuth2 client to access the information under certain IdPs.
 
+    This model is no longer in use.
+
+    TODO: BOM-1270: This model and table will be removed post-Juniper
```

Comment:
@jinder1s: I forgot that your instructions need to be slightly more involved, because the old OAuth Application will be DOP, so someone first needs to create the DOT equivalent, and then create the Application Access with the filter, and also create a record in Restricted Application for the DOT application.

The first steps regarding DOP will need to review to use SQL from the DB, because admin will be gone. You could point them to https://github.com/edx/edx-platform/blob/372d2e927c9eec2a5342ab120f84cc290578aa90/openedx/core/djangoapps/oauth_dispatch/docs/how_tos/testing_manually.rst for additional help.

Reply:
I would remove this model entirely and include instructions on dealing with it elsewhere, maybe the Juniper release page. As long as it sticks around, it will still rely on the DOP `Client` model, preventing us from removing DOP as a dependency entirely.

Reply:
Okay, after speaking to @jinder1s , there are good reasons to leave the model around. My proposal is then to remove the foreign key relation first so that we can drop the client library as an import/dependency for edx-platform.

Reply:
I decided to remove the test, but will not run makemigration in this PR due to recommendation from https://openedx.atlassian.net/l/c/Y1VJdjLs